apps/angular/bug-cd/src/app/main-navigation.component.ts

```diff
@@ -40,23 +41,15 @@ export class NavigationComponent {
 @Component({
   standalone: true,
   imports: [NavigationComponent, NgIf, AsyncPipe],
-  template: `
-    <ng-container *ngIf="info$ | async as info">
-      <ng-container *ngIf="info !== null; else noInfo">
-        <app-nav [menus]="getMenu(info)" />
-      </ng-container>
-    </ng-container>
-
-    <ng-template #noInfo>
-      <app-nav [menus]="getMenu('')" />
-    </ng-template>
-  `,
+  template: ` <app-nav [menus]="(menus$ | async) ?? []" /> `,
   host: {},
 })
 export class MainNavigationComponent {
   private fakeBackend = inject(FakeServiceService);
 
-  readonly info$ = this.fakeBackend.getInfoFromBackend();
+  readonly menus$ = this.fakeBackend
+    .getInfoFromBackend()
+    .pipe(map((info) => this.getMenu(info ?? '')));
```

Comment:
you can now use the signal API, but the solution is correct

Reply:
to be fair, i thought about converting this http response observable toSignal,  but used rxjs since i had to transform the value by sending to getMenu function to get the result value for the template